mongo/options/createcollectionoptions.go

```diff
@@ -28,18 +30,35 @@ func (d *DefaultIndexOptions) SetStorageEngine(storageEngine interface{}) *Defau
 
 // TimeSeriesOptions specifies options on a time-series collection.
 type TimeSeriesOptions struct {
-	// Name of the top-level field to be used for time. Inserted documents must have this field,
+	// TimeField is the top-level field to be used for time. Inserted documents must have this field,
 	// and the field must be of the BSON UTC datetime type (0x9).
 	TimeField string
 
-	// Optional name of the top-level field describing the series. This field is used to group
+	// MetaField is the name of the top-level field describing the series. This field is used to group
 	// related data and may be of any BSON type, except for array. This name may not be the same
-	// as the TimeField or _id.
+	// as the TimeField or _id. This field is optional.
 	MetaField *string
 
-	// Optional string specifying granularity of time-series data. Allowed granularity options are
-	// "seconds", "minutes" and "hours".
+	// Granularity is the granularity of time-series data. Allowed granularity options are
+	// "seconds", "minutes" and "hours". This field is optional.
 	Granularity *string
+
+	// BucketMaxSpanSeconds is the maximum range of time values for a
+	// bucket, in seconds. The server will enforce that
+	// "bucketRoundingSeconds" is equal to "bucketMaxSpanSeconds", and that
+	// both are strictly positive. If a user sets one value, they must set
+	// both. If a user sets these values, they cannot also set
+	// "Granularity". This field is optional.
+	BucketMaxSpanSeconds *time.Duration
```

Comment:
```suggestion
	BucketMaxSpan *time.Duration
```

Including `Seconds` in the name may surprise users that do not include the time unit. E.g. passing only `5` is interpreted as nanoseconds:

```go
tsopts := options.TimeSeries().SetTimeField("myfield").SetBucketRoundingSeconds(5)
fmt.Printf("BucketRoundingSeconds is: %v\n", tsopts.BucketRoundingSeconds)
// Prints: "BucketRoundingSeconds is: 5ns"
```

The `FindOptions.MaxTime` option follows the same pattern. The server option is named `MaxTimeMS`, but the Go option is named `MaxTime` and accepts a `time.Duration`.